src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts

```diff
@@ -327,7 +327,20 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 				componentProperties: {
 					title: localize('tableDesigner.columnTypeTitle', "Type"),
 					width: 100,
-					values: columnTypes
+					values: columnTypes,
+					isEditable: true
+				}
+
+			}, {
+				componentType: 'dropdown',
```

Comment:
I thought we discussed this, you shouldn't need to modify anything in core. this should be done in STS as a sql server specific logic.

also, you wanted to scope this PR to just be the file move, right?

Reply:
Yes, we discussed adding a new column which is what this is doing. Since you already have context about the task and the change in core is minimal, I added it in the same PR.

Reply:
1. we discussed offline that the new property will be added in STS not here, but I guess it is also ok to be added here, this pattern should also be applicable to other providers.
2. you can't have duplicate property names, please define a new one.
3. I don't think you have the correct settings, correct setting should be  the one in the properties view should be readonly the one in the table is editable
4. you need to add the new compound data type to the  columns list so that it gets displayed.

Reply:
1 & 2.  I asked you offline and you said to add it in core. I also double checked by asking what name of the column you want and you said the same name. Could you please clarify what exactly you're envisioning in writing? It seems I often get different instructions online vs offline.
3. As discussed offline, the newly added property is editable in the table view and not in the properties view. We only show the normal dropdown in properties view, following our discussion offline when we looked at SSDT.
4. Yes, that work needs STS PR's too, and will probably bundle in another PR.

Reply:
for 1, I am pretty sure I said for this feature no table designer related change should happen in core, you should overwrite the default columns in STS, but as I mentioned in my comments above, it is okay to define it in core since the pattern can be helpful for other providers as well.

for 2, I did say i want the same display name (title), but you have to have unique property names, otherwise you won't be able to get the data properly.

for 3, what you described is correct, but the code is not doing what you described. showInPropertiesView and isEditable are both false.

for 4, no, if we are adding this new data type property in core, you don't have to add it in STS, you can modify the visible properties for table columns table in this file.

![image](https://user-images.githubusercontent.com/13777222/156274430-8ef5bd7a-c96c-46e8-a262-0442b799b168.png)

Reply:
for 4, you need to add the new property to the default displayProperties and remove the other 4 properties from it. it is defined here, let me know if you have any questions.
![image](https://user-images.githubusercontent.com/13777222/156274995-49c68377-8e26-4fc0-b744-aa34569c25ec.png)